External/FEXCore/Source/Interface/Core/Core.cpp

```diff
@@ -621,6 +622,47 @@ namespace FEXCore::Context {
     return Thread;
   }
 
+  void Context::DestroyThread(FEXCore::Core::InternalThreadState *Thread) {
+    // remove new thread object
+    {
+      std::lock_guard<std::mutex> lk(ThreadCreationMutex);
+
+      auto It = std::find(Threads.begin(), Threads.end(), Thread);
+      LogMan::Throw::A(It != Threads.end(), "Thread wasn't in Threads");
+
+      Threads.erase(It);
+    }
+
+    if (std::this_thread::get_id() == Thread->ExecutionThread.get_id()) {
+      // To be able to delete a thread from itself, we need to detached the std::thread object
+      Thread->ExecutionThread.detach();
+    }
+    delete Thread;
+  }
+
+  void Context::DeleteForkedThreads(FEXCore::Core::InternalThreadState *ExceptForThread) {
+    // This function is called after fork
+    // We need to cleanup some of the thread data that is dead
+    for (auto &DeadThread : Threads) {
+      if (DeadThread == ExceptForThread) {
+        continue;
+      }
+
+      // Setting running to false ensures that when they are shutdown we won't send signals to kill them
+      DeadThread->RunningEvents.Running = false;
+
+      // Clean up thread state
+      delete DeadThread;
```

Comment:
I'm really not sure about this. None of our tests actually hit this.

Assuming any program is evil enough to fork while it has threads, the `std::thread` thread is going to be desynced with reality. The threads should still be marked as `joinable()` which means deleting the object will `std::terminate()`

But they can't actually be joined or detached.

I suspect the only sane thing to do here is to do something evil to deallocate the std::thread without actually deconstructing it.

Reply:
It's fine, as long as we join. See comments in followup commit.